packages/sdk-core/src/model/report/BacktraceReport.ts

```diff
@@ -1,7 +1,52 @@
 import { BacktraceAttachment } from './BacktraceAttachment';
+import { BacktraceErrorType } from './BacktraceErrorType';
 
-export interface BacktraceReport {
-    readonly attributes: Record<string, unknown>;
-    readonly attachments: BacktraceAttachment[];
-    toData(): Record<string, unknown>;
+export class BacktraceReport {
+    /**
+     * Report classifiers
+     */
+    public readonly classifiers: string[] = [];
+    /**
+     * Report annotations
+     */
+    public readonly annotations: Record<string, object> = {};
+    /**
+     * Report stack trace
+     */
+    public readonly stackTrace: string;
+
+    /**
+     * Report message
+     */
+    public readonly message: string;
+
+    /**
+     * Report inner errors
+     */
+    public readonly innerReport: unknown[] = [];
+
+    constructor(
+        public readonly data: Error | string,
+        public readonly attributes: Record<string, unknown> = {},
+        public readonly attachments: BacktraceAttachment[] = [],
+    ) {
+        let errorType: BacktraceErrorType = 'Exception';
+        if (data instanceof Error) {
+            this.annotations['error'] = data;
+            this.classifiers = [data.name];
+            this.message = data.message;
+            this.stackTrace = data.stack ?? '';
+            if (data.cause) {
+                this.innerReport.push(data.cause);
+            }
+        } else {
+            this.message = data;
+            this.stackTrace = new Error().stack ?? '';
+            errorType = 'Message';
+        }
+
+        if (!this.attributes['error.type']) {
+            this.attributes['error.type'] = errorType;
+        }
+    }
```

Comment:
the use case for this for our clients is to write 

try {
} catch(err) {
if(something) {
const report =new BacktraceReport(err);
client.send(report);
}
} 

if we generate additional static methods like fromError it might generate a lot of confusion for our users